File: languages/fortran/projects/01_grade_report.f90
! Objetivo: montar um mini projeto de relatorio de notas.
! Como executar:
!   powershell -ExecutionPolicy Bypass -File .\scriptsuild-fortran-module.ps1 -Source .\languagesortran\projects_grade_report.f90
! Conceitos treinados: arrays, funcoes internas, maxval, minval.
! Próximo exercício: adaptar para ler notas do teclado.

program grade_report
    implicit none
    real, dimension(5) :: notas

    notas = (/ 8.0, 7.5, 9.0, 6.5, 8.8 /)

    print *, "Relatorio da turma"
    print *, "Notas:", notas
    print *, "Media:", media(notas)
    print *, "Maior nota:", maxval(notas)
    print *, "Menor nota:", minval(notas)

contains

    real function media(v)
        implicit none
        real, dimension(:), intent(in) :: v
        media = sum(v) / real(size(v))
    end function media

end program grade_report

File: languages/php/mini_projects/01_grade_report.php
<?php
/*
Objetivo: mini projeto simples para relatório de notas.
Como executar: php .\languages\php\mini_projects\01_grade_report.php
Conceitos treinados: arrays, funções, condição.
Próximo exercício: persistir em arquivo JSON.
*/

$students = [
    ['nome' => 'Ana', 'nota' => 8.0],
    ['nome' => 'Bruno', 'nota' => 5.5],
    ['nome' => 'Carla', 'nota' => 9.1],
];

foreach ($students as $student) {
    $status = $student['nota'] >= 6.0 ? 'Aprovado' : 'Reprovado';
    echo $student['nome'] . ' | ' . $student['nota'] . ' | ' . $status . PHP_EOL;
}
